std/range/package.d

```diff
@@ -1841,9 +1841,9 @@ if (isInputRange!(Unqual!Range) &&
         if (hasSlicing!R)
         {
             assert(i <= j, "Invalid slice bounds");
-            assert(j - i <= length, "Attempting to slice past the end of a "
+            assert(j <= length, "Attempting to slice past the end of a "
                 ~ Take.stringof);
-            return source[i .. j - i];
+            return source[i .. j];
```

Comment:
Oops. Not sure what I was thinking when I wrote this. Kind of sad it took 2 years to catch it :/